src/Model/CompositeTypeGo.cs

```diff
@@ -26,6 +27,28 @@ public class CompositeTypeGo : CompositeType
 
         public bool PreparerNeeded = false;
 
+        public IEnumerable<CompositeType> DerivedTypes => CodeModel.ModelTypes.Where(t => t.DerivesFrom(this));
+
+        public IEnumerable<CompositeType> SiblingTypes
+        {
+            get
+            {
+                var st = (BaseModelType as CompositeTypeGo).DerivedTypes;
+                if (BaseModelType.BaseModelType != null && BaseModelType.BaseModelType.IsPolymorphic)
+                {
+                    st = st.Union((BaseModelType as CompositeTypeGo).SiblingTypes);
+                }
+                return st;
+            }
+        }
+
+        public bool HasPolymorphicFields => AllProperties.Any(p => (p.ModelType is CompositeType && (p.ModelType as CompositeTypeGo).IsPolymorphic) 

```

Comment:
Too much shortening! Lets give this a real method body.

Reply:
Made all this more readable